### PR TITLE
do not allow to create a new branch if the entered branch name is already in use

### PR DESCRIPTION
**Pull request by DavidFeldhoff**
I receive the remote refs using the API to make sure that no fetch is missing. And I don't want to execute `git.fetch` as the user doesn't explicitly asked for it, so I'm just reading stuff and not changing any state.

### CODE REVIEW

File: src/api/azdevops-api.ts [hunk 1]
```diff
@@ -438,10 +439,20 @@ export class WorkItemTreeItem extends vscode.TreeItem {
...
+            const localRefs: string[] = repo.state.refs.filter(ref => ref.name !== undefined && ref.type !== RefType.RemoteHead).map(ref => ref.name!)
+            const existingRefs = remoteRefs.concat(localRefs);
             const gitPrefix = vscode.workspace.getConfiguration('git').get('branchPrefix', "");
             let newBranch = await vscode.window.showInputBox({
                 prompt: "Please enter the name of the new branch",
-                value: `${gitPrefix !== "" ? `${gitPrefix}` : ""}${useWorkitemIdInBranchName() ? this.wiId : ""}`
+                value: `${gitPrefix !== "" ? `${gitPrefix}` : ""}${useWorkitemIdInBranchName() ? this.wiId : ""}`,
+                validateInput: (value: string) => {
+                    const existingref = existingRefs.find(refName => refName.toLowerCase() === value.toLowerCase());
+                    if (existingref !== undefined) {
```
Comment (tfenster) [inline src/api/azdevops-api.ts:451]: if (existingref)

File: src/api/azdevops-api.ts [hunk 2]
```diff
@@ -438,10 +439,20 @@ export class WorkItemTreeItem extends vscode.TreeItem {
...
+                        return `Branch ${existingref} already exists. Please choose another one`
```
Comment (tfenster) [inline src/api/azdevops-api.ts:452]: missing ;

File: src/api/azdevops-api.ts [hunk 3]
```diff
@@ -438,10 +439,20 @@ export class WorkItemTreeItem extends vscode.TreeItem {
     public async createBranch() {
         const repo = getGitExtension().getRepo();
         if (repo) {
+            let remoteRefs: string[] = await getRemoteRefs(this.parent.parent.url);
+            const localRefs: string[] = repo.state.refs.filter(ref => ref.name !== undefined && ref.type !== RefType.RemoteHead).map(ref => ref.name!)
```
Comment (tfenster) [inline src/api/azdevops-api.ts:443]: missing ;

File: src/api/azdevops-api.ts [hunk 4]
```diff
@@ -438,10 +439,20 @@ export class WorkItemTreeItem extends vscode.TreeItem {
...
             let newBranch = await vscode.window.showInputBox({
                 prompt: "Please enter the name of the new branch",
-                value: `${gitPrefix !== "" ? `${gitPrefix}` : ""}${useWorkitemIdInBranchName() ? this.wiId : ""}`
+                value: `${gitPrefix !== "" ? `${gitPrefix}` : ""}${useWorkitemIdInBranchName() ? this.wiId : ""}`,
+                validateInput: (value: string) => {
+                    const existingref = existingRefs.find(refName => refName.toLowerCase() === value.toLowerCase());
+                    if (existingref !== undefined) {
+                        return `Branch ${existingref} already exists. Please choose another one`
+                    }
+                    return undefined
```
Comment (tfenster) [inline src/api/azdevops-api.ts:454]: missing ;

**Review by tfenster — CHANGES_REQUESTED** [review]
I actually don't feel comfortable to nitpick on that level in an open source, volunteer project, but I found a couple of small details that can be improved

**Comment by DavidFeldhoff** [comment]
I enabled now ES-Lint for me as well and fixed all the warnings coming up 😉

**Review by tfenster — DISMISSED** [review]

**Comment by DavidFeldhoff** [comment]
Hm.. After merigng there were further ES-Lint issues which I just fixed. But now I need once again your review, please ^^

**Review by tfenster — APPROVED** [review]

**Pull request merged**